extensions/interactions/CodeRepl/directives/CodeRepl.js

```diff
@@ -21,16 +21,29 @@
  */
 oppia.directive('oppiaInteractiveCodeRepl', [
   'HtmlEscaperService', 'codeReplRulesService', 'UrlInterpolationService',
-  function(HtmlEscaperService, codeReplRulesService, UrlInterpolationService) {
+  'EVENT_NEW_CARD_AVAILABLE',
+  function(HtmlEscaperService, codeReplRulesService, UrlInterpolationService,
+      EVENT_NEW_CARD_AVAILABLE ) {
     return {
       restrict: 'E',
       scope: {
-        onSubmit: '&'
+        onSubmit: '&',
+        getLastAnswer: '&lastAnswer'
       },
       templateUrl: UrlInterpolationService.getExtensionResourceUrl(
         '/interactions/CodeRepl/directives/' +
         'code_repl_interaction_directive.html'),
       controller: ['$scope', '$attrs', function($scope, $attrs) {
+        if ($scope.getLastAnswer()) {
```

Comment:
This is executed only at the beginning, so if during an interaction, he submits answers, I don't think it would disable the interaction.

Reply:
Ah, OK. I think it would be better to do something more robust, though -- e.g. in the future we might decide to repaint the directive midway.

Can we say something like "if this is not the last card" (or "active card", if that's what active means) instead? That seems more robust and does not rely on other implementation details.

Reply:
Ok, then shall I inject PlayerTranscriptsService into the interactions, then? As it has some functions for those operations.

Reply:
OK, what I am trying now is, in supplemental card directive I set lastAnswer as false if it is last card and else I give it the answer. How about this?